dcos_test_utils/gce.py

```diff
@@ -0,0 +1,284 @@
+      accessConfigs:
+      - name: External NAT
+        type: ONE_TO_ONE_NAT
+    metadata:
+      items:
+      - key: ssh-keys
+        value: {ssh_user}:{ssh_public_key}
+"""
+
+# template for a network resource in a gce deployment
+NETWORK_TEMPLATE = """
+type: compute.v1.network
+name: {name}
+properties:
+  autoCreateSubnetworks: True
+"""
+
+# template for an instance group manager resource in a gce deployment
+MANAGED_INSTANCE_GROUP_TEMPLATE = """
+type: compute.v1.instanceGroupManager
+name: {name}
+metadata:
+  dependsOn:
+  - {instance_template_name}
+properties:
+  baseInstanceName: vm
+  instanceTemplate: global/instanceTemplates/{instance_template_name}
+  zone: {zone}
+  targetSize: {size}
+"""
+
+# template for a firewall in the network of a gce deployment
+FIREWALL_TEMPLATE = """
+type: compute.v1.firewall
+name: {name}
+metadata:
+  dependsOn:
+  - {network}
+properties:
+  description: allow all ports
+  network: global/networks/{network}
+  sourceRanges:
+  - 0.0.0.0/0
+  allowed:
+  - IPProtocol: tcp
+  - IPProtocol: udp
+  - IPProtocol: icmp
+  - IPProtocol: sctp
+"""
+
+
+# Function decorator that adds detail to potential googleapiclient.errors.HttpError exceptions with code 404 or 409
+def catch_http_exceptions(f):
+    @wraps(f)
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if e.resp.status == 404:
+                log.exception("The resource you are trying to access doesn't exist")
+            elif e.resp.status == 409:
+                log.exception('''The specified resources exist and might be under an active operation
+                                   (operation conflict)''')
+            raise e
+
+    return handle_exception
+
+
+class GceWrapper:
+    @catch_http_exceptions
+    def __init__(self, credentials_dict: dict, credentials_path: str):
+        credentials = ServiceAccountCredentials.from_json_keyfile_name(
+            credentials_path, scopes='https://www.googleapis.com/auth/cloud-platform')
+        self.compute = discovery.build('compute', 'v1', credentials=credentials)
+        self.deployment_manager = discovery.build('deploymentmanager', 'v2', credentials=credentials)
+        self.project_id = credentials_dict['project_id']
+
+    @catch_http_exceptions
+    def get_instance_info(self, name: str, zone: str):
+        """ Returns the dictionary representation of a GCE instance resource. For details on the contents of this
+        resource,see https://cloud.google.com/compute/docs/reference/latest/instances
+        """
+        response = self.compute.instances().get(project=self.project_id, zone=zone, instance=name).execute()
+        log.debug('get_instance_info response: ' + str(response))
+        return response
+
+    @catch_http_exceptions
+    def list_group_instances(self, group_name: str, zone: str) -> typing.Iterator(dict):
```

Comment:
Shouldn't this be `typing.Iterator[dict]`? I got an error from the tests if I didn't change them from () to []. Idk why it would work for others -- my type hinting skillz are weak.